### ISSUE TITLE
[Feature request] Handle running processes automatically with a explicit flag

### ISSUE THREAD

**Issue by bartier** [elastic/rally#922]
When I run esrally with default settings (i.e --on-error=continue), I can 'cancel' the esrally process using `CTRL+C` in my terminal. Maybe I would like to make a change and restart esrally.

The problem is: the next time I run esrally, it won't be able to start a race because of the old esrally process:
```
    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

[ERROR] Cannot race. There are other Rally processes running on this machine (PIDs: [4122, 4123, 4124, 4172, 4234, 4235]) but only one Rally benchmark is allowed to run at the same time. Please check and terminate these processes and retry again.

Getting further help:
*********************
* Check the log files in /home/vitor/.rally/logs for errors.
* Read the documentation at https://esrally.readthedocs.io/en/1.4.0/
* Ask a question on the forum at https://discuss.elastic.co/c/elasticsearch/rally
* Raise an issue at https://github.com/elastic/rally/issues and include the log files in /home/vitor/.rally/logs.

-------------------------------
[INFO] FAILURE (took 0 seconds)
-------------------------------
```

Steps to reproduce:

1. Run `esrally --distribution-version=7.6.0 --track percolator`
2. In your terminal, cancel the command with `CTRL+C`
![image](https://user-images.githubusercontent.com/18057391/75631570-c71c4580-5bd2-11ea-89ea-c490e0e140f0.png)

3. Try to run `esrally --distribution-version=7.6.0 --track percolator` again.
4. You will see another processes running in background. You need to manually kill them.

This issue is to suggest a flag in esrally that could kill automatically these running process to help the usability. The default could be the warning with a flag suggestion, something like `--kill-running-process` or something to automatically deal with it.

I'm not sure if this is appropriate feature to have in esrally or if the processes are entirely controlable by the SO.

**Comment by danielmitterdorfer** [comment]
Thanks for your interest in Rally and suggesting this. Originally Rally has cleaned up any leftover processes itself but we had several surprised users and changed this a while ago in https://github.com/elastic/rally/issues/383 to the current behavior that you are observing. 

In the mid term we are planning make changes to Rally which make such a flag hopefully obsolete but meanwhile I think such a flag could be useful.

**Comment by bartier** [comment]
@danielmitterdorfer Thanks for the answer. I'd like to work on this if possible (I'm not sure when haha), 

By the way, I'm not sure if this config should be overriten in `rally.ini`. Actually, I've made some tests and I could make work with a flag when running rally in the command line. 

If I set the below configuration in `rally.ini`, this doesn't reflect in the application. Is there any idea why? 
![image](https://user-images.githubusercontent.com/18057391/78464608-d276fb00-76c1-11ea-8d5e-f8b648514dd1.png)

I made the following changes:

Add the argument to parser (Ok to me)
```
        p.add_argument(
            "--kill-running-processes",
            action="store_true",
            default=False,
            help="If any processes is running, it is going to kill them and allow rally to continue to run."
        )
```

Add `cfg.add(config.Scope.applicationOverride, "system", "kill.running.processes", args.kill_running_processes)` in the `main()`.

The 2 changes above only works with  explicitly  command flag , the `rally.ini` does not respect the value. Example:
`rally.py --distribution-version=7.6.2 --kill-running-processes` reflects the configuration i `cfg._opts`
`rally.py --distribution-version=7.6.2 ` and `rally.ini` with `kill.running.processes` to `true` does not reflect the configuration i `cfg._opts`
